### ISSUE TITLE
Discuss removing recommendation in menubutton pattern to add/remove aria-expanded="true

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/aria-practices#697]
For menu buttons and menu bars there is a different treatment of aria-expanded. 
For the [menu button](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-links.html), there is no aria-expanded on page load and then only added when it's open. So it doesn't toggle between true and false ... instead the entire attribute is added and removed.  This apparently is to [reduce chattiness](https://github.com/w3c/aria-practices/issues/326#issuecomment-295584897)

This process of adding and removing the entire attribute on expand/collapse has spotty support in JAWS, VO and NVDA. It is also tough to explain to devs... They will ask why add and remove the entire attribute when there is true/false to toggle in aria-expanded. 

On the other hand, the [navigation menu](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html) has aria-expanded=false on page load. So the two examples are inconsistent.  

 It seems like we could have a more elegant and consistent solution for aria-expanded on menus. 

Maybe I should file a but against aria-haspop over on the aria 1.1 to allow an extra attribute to swap out for menus so it doesn't have the redundancy but doesn't require the clunky adding and removing the full attribute to show expanded state.... 

Or maybe AT should not announce collapsed when an aria-popup is present??

Anyway, adding removing the entire attribute and value on show/hide smells like a hack to address AT issues, or to overcome a limitation of wai-aria.  

This is the current pertinent  language in the button example:

> "Added when the menu is open.Indicates that the menu is displayed and that activating the menu button closes the menu. The aria-expanded attribute is removed when the menu is closed. Included to support touch devices where screen reader users can touch the menu button when the menu is displayed. Keyboard users cannot focus the menu button when the menu is open.

**Comment by patrickhlauke** [comment]
removing it from where? this lacks a bit of context...

**Comment by DavidMacDonald** [comment]
sorry @patrickhlauke, was filling in issue and hit enter... its there now.

**Comment by mcking65** [comment]
@DavidMacDonald, not including aria-expanded when the menu is not displayed is optional behavior. The way we worded this in the pattern is:

>When the menu is displayed, the element with role button has aria-expanded set to true. When the menu is hidden, it is recommended that aria-expanded is not present. If aria-expanded is specified when the menu is hidden, it is set to false.

If we updated the states and properties table on the menubutton example pages to include similar language, would that be sufficient?

I don't see a need to change the ARIA spec.

**Comment by brennanyoung** [comment]
If add/remove is preferred, a proper boolean attribute is The Right Thing. But AFAIK wai-aria has no true boolean attributes. The add/remove mechanism of true boolean attributes are already a source of many cognitive errors (because `someBooleanAttribute="false"` evaluates -counter-intuitively- the same as `someBooleanAttribute="true"`)

IMO an attribute with possible values true/false should have those values set explicitly, rather than removing it to suggest 'false'. Inconsistencies are inevitable if an attribute becomes optional in a some states and not others. 

Reducing doubt amongst those consulting the examples should override vague concerns about 'chattiness'. The 'chattiness' problem is for the AT vendors to handle (if indeed it is a problem at all). 

But... consistency first. Whichever is the preferred mechanism, let's do it the same way in all examples and recommendations - and if we're going with add/remove, perhaps provide a note about why this might be counter-intuitive. Otherwise it will become a FAQ.
